### PR TITLE
Backport (1.6.x): DBPW - HanaDB to v5; enable AutoMTLS; remove `newdbplugin` package

### PR DESCRIPTION
**Pull request by pcman312**
Backports:
- https://github.com/hashicorp/vault/pull/10142 - Migrate HanaDB to DBv5 interface
- https://github.com/hashicorp/vault/pull/10216 - Remove `newdbplugin` package
- https://github.com/hashicorp/vault/pull/10220 - Enable AutoMTLS, temporarily disabling couchbase, elasticsearch, & mongodbatlas
- https://github.com/hashicorp/vault/pull/10222 - Add couchbase, elasticsearch, & mongodbatlas back in

### CODE REVIEW

File: go.mod
Comment (tvoran) [inline go.mod:86]: Should these be tracking the release branches of each plugin repo? i.e. something more like
```suggestion
	github.com/hashicorp/vault-plugin-database-couchbase v0.2.1
	github.com/hashicorp/vault-plugin-database-elasticsearch v0.6.1
	github.com/hashicorp/vault-plugin-database-mongodbatlas v0.2.1
```

Comment (pcman312) [reply go.mod:86]: No because that's not what the original PR did. Those tags ended up being created afterwards. They're effectively the same, but I didn't want to hold up that PR to wait for tagging of the repos.

Comment (tvoran) [reply go.mod:86]: Ok, since this contains so many other changes, I'm ok with updating the tagging and versions after this.

Comment (calvn) [reply go.mod:86]: This is fine, but you'd have to do a follow-up PR (against the release branch) after this is merged to use the actual release tags as Theron is suggesting.

**Review by calvn — APPROVED** [review]

**Review by tvoran — APPROVED** [review]

File: go.mod
Comment (pcman312) [reply go.mod:86]: I can update this to use the release tags now.

**Pull request merged**
